pyrefly/lib/test/simple.rs

```diff
@@ -1674,3 +1674,17 @@ def f(condition: bool):
     assert_type(x, A | type[A])
     "#,
 );
+
+testcase!(
+    test_nested_list_comp,
+    r#"from typing import assert_type
+assert_type([a for a in [b for b in ["a", "b"]]], list[str])
+"#,
+);
+
+testcase!(
+    test_incomplete_nested_list_comp,
+    r#"# Issue https://github.com/facebook/pyrefly/issues/455
```

Comment:
why is there a link to the GH issue here? I imagine it's not intentional?

Reply:
I saw an example of issue reference [here](https://github.com/facebook/pyrefly/blob/main/pyrefly/lib/test/simple.rs#L1607), so I decided to left reference too. It might be helpful to understand why so uncommon test case has been written. Should I remove it?

Reply:
I think it's fine